### PR TITLE
Create a clang-format file. Update all source files

### PR DESCRIPTION
**Pull request by umar456**
Use clang format to enforce style.

### CODE REVIEW

**Comment by umar456** [comment]
Let me change it so that they are on separate commits

**Comment by umar456** [comment]
@pavanky See https://github.com/arrayfire/arrayfire/pull/2390/commits/9c894713737ddb36a0e63d0e68ea0d9baef008e4

**Comment by umar456** [comment]
The only difference between the three format files is the Standard line on line 141. In examples and tests is Cpp03 vs Cpp11 in src

**Comment by 9prady9** [comment]
At the moment, we are not using any tool to enforce this style as far as I can gather from the PR. It might be better to update `https://github.com/arrayfire/arrayfire/blob/master/CONTRIBUTING.md` file to reflect the requirement of running clang-format before submitting a code change.

File: examples/.clang-format [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  AfterObjCDeclaration: false
+  AfterStruct:     false
+  AfterUnion:      false
+  AfterExternBlock: false
+  BeforeCatch:     false
+  BeforeElse:      false
+  IndentBraces:    false
+  SplitEmptyFunction: true
+  SplitEmptyRecord: true
+  SplitEmptyNamespace: true
```
[the remark this answers is not in the document]
Comment (umar456) [reply examples/.clang-format]: Fixed. It was being ignored because of another property. Changed it to false

**Comment by umar456** [comment]
It looks for the first .clang-format file and uses that. It doesn't continue going up the project tree.

**Comment by 9prady9** [comment]
One more query, how about some instructions in `CONTRIBUTING.md` about this style guide. Also, we should put some help on how to setup `clang-format`  and use it. It might not be difficult to setup and use it, but it definitely helps a lot and be quick for a developer if we have these written down somewhere.

**Pull request merged**